deploy/apex-client/base/daemonset.yaml

```diff
@@ -0,0 +1,118 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: apex
+spec:
+  updateStrategy:
+    type: OnDelete
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: apex-client
+      app.kubernetes.io/instance: apex-client
+      app.kubernetes.io/name: apex-client
+      app.kubernetes.io/part-of: apex
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/part-of: apex
+    spec:
+      hostNetwork: true
+      hostAliases:
+        - ip: "127.0.0.1"
+          hostnames:
+            - "auth.apex.local"
+            - "api.apex.local"
+            - "apex.local"
+      serviceAccount: apex
+      serviceAccountName: apex
+      tolerations:
+        - key: node-role.kubernetes.io/control-plane
+          operator: Exists
+          effect: NoSchedule
+        - key: node-role.kubernetes.io/master
+          operator: Exists
+          effect: NoSchedule
+      initContainers:
+        - name: cert-setup
+          image: quay.io/apex/apex:latest
+          imagePullPolicy: Always
+          command:
+            - /bin/sh
+            - -c
+            - |
+              mkdir -p /etc/apex/.certs; base64 -d /etc/apex-client-secret/cert > /etc/apex/.certs/rootCA.pem
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - name: host-apex
+              mountPath: "/etc/apex"
+              mountPropagation: Bidirectional
+            - name: apex-client-secret
+              mountPath: "/etc/apex-client-secret"
+              readOnly: true
+      containers:
+        - name: apex-client
+          image: quay.io/apex/apex:latest
+          imagePullPolicy: Always
+          env:
+            - name: USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: apex-client-secret
+                  key: username
+                  optional: false
+            - name: PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: apex-client-secret
+                  key: password
+                  optional: false
+            - name: URL
+              valueFrom:
+                secretKeyRef:
+                  name: apex-client-secret
+                  key: url
+                  optional: false
+            - name: AUTH_URL
+              valueFrom:
+                secretKeyRef:
+                  name: apex-client-secret
+                  key: auth_url
+                  optional: false
+          command:
+            - /bin/sh
+            - -c
+            - |
+              CAROOT=/etc/apex/.certs ./mkcert -install
+              /apexd --username=$USERNAME --password=$PASSWORD $URL
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/sh", "-c", "ip link delete wg0"]
```

Comment:
If you remove the pod from any Kuberntes node, wireguard interface will still exist on that node, so it needs to be cleaned up manually. Assumption here is that if you are removing the pod you want to remove the device from the wireguard networking, so the cleanup is required, otherwise the device will like a dangling device in the network.

Reply:
thanks